modules/javafx.graphics/src/main/java/javafx/css/SimpleSelector.java

```diff
@@ -321,8 +376,8 @@ private boolean matchStyleClasses(StyleClassSet otherStyleClasses) {
     @Override public int hashCode() {
         int hash = 7;
         hash = 31 * (hash + name.hashCode());
-        hash = 31 * (hash + styleClassSet.hashCode());
-        hash = 31 * (hash + styleClassSet.hashCode());
+        hash = 31 * (hash + selectorStyleClassNames.hashCode());
+        hash = 31 * (hash + selectorStyleClassNames.hashCode());
         hash = (id != null) ? 31 * (hash + id.hashCode()) : 0;
         hash = 31 * (hash + pseudoClassState.hashCode());
         return hash;
```

Comment:
Is it worth caching the hash code for this class?

Reply:
I put a debug print in to this method, and then started JFXCentral application (which is quite big) -- it's never called, so doesn't look like it is used anywhere.  Selectors are most often stored in `List`s, didn't find any `Map`s or `Set`s that use it as a key.